src/publicTypes.ts

```diff
@@ -136,4 +136,56 @@ export type NamehashOptions = {
 };
 
 export const NamehashOptionsDefault = {format: 'hex', prefix: true} as const;
+
+export enum DnsRecordType {
+A="A",
+AAAA="AAAA",
+AFSDB="AFSDB",
+APL="APL",
+CAA="CAA",
+CDNSKEY="CDNSKEY",
+CDS="CDS",
+CERT="CERT",
+CNAME="CNAME",
+CSYNC="CSYNC",
+DHCID="DHCID",
+DLV="DLV",
+DNAME="DNAME",
+DNSKEY="DNSKEY",
+DS="DS",
+EUI48="EUI48",
+EUI64="EUI64",
+HINFO="HINFO",
+HIP="HIP",
+HTTPS="HTTPS",
+IPSECKEY="IPSECKEY",
+KEY="KEY",
+KX="KX",
+LOC="LOC",
+MX="MX",
+NAPTR="NAPTR",
+NS="NS",
+NSEC="NSEC",
+NSEC3="NSEC3",
+NSEC3PARAM="NSEC3PARAM",
+OPENPGPKEY="OPENPGPKEY",
+PTR="PTR",
+RP="RP",
+RRSIG="RRSIG",
+SIG="SIG",
+SMIMEA="SMIMEA",
+SOA="SOA",
+SRV="SRV",
+SSHFP="SSHFP",
+SVCB="SVCB",
+TA="TA",
+TKEY="TKEY",
+TLSA="TLSA",
+TSIG="TSIG",
+TXT="TXT",
+URI="URI",
+ZONEMD="ZONEMD",
+}
+
+export type DnsRecord = {type: DnsRecordType, ttl: number, value: string}
```

Comment:
So I was thinking a bit more on use cases. @bogdan from our last conversation we concluded that this structure probably reflects UI of management sites.
Since we don't allow different TTLs for records of the same type, will it make sense to design UIs like this at all? Do we want to assume that there is such data structure that allows it? Probably it makes sense to make it even more explicit, that UIs shouldn't be designed the way that would allow setting different TTLs.

Imagine the following design:
```
Type | TTL | Data
-----|-----|-------------------------------
A    | 128 | 10.0.0.1
     |     | 10.0.0.2
     |     | [add more]
-----|-----|-------------------------------
AAAA | 128 | fdda:5cc1:23:4::1d  
     |     | fdda:5cc1:23:4::1f
     |     | [add more]
```
I'm trying to depict here that a user won't be able to specify other values in blank fields (like another TTL), but they'll just be able to add another value to the `Data` column.

As of other use-cases, I can't come up with anything else atm, when a user would need a key-value structure instead of a list of DoH-like answers.

Reply:
We can not control the UI. I think we have to stick with `DnsRecord[]` format in a library. But I agree that your UI proposal looks good. Maybe this code will be useless for that UI, but I still think it is fine to implement our algorithm because it requires us answers on how to handle edge cases.